Chore/222 investigate igraph performance (#305)
Deltares/ra2ce: docs/collaboration/implementation_decisions.rst

However, this did not seem to be a valid alternative as the accuracy and simplicity of geopy outweights the benefit of replacing it with similar extensive logic as can be seen in this `example <https://autogis-site.readthedocs.io/en/2019/notebooks/L2/calculating-distances.html>`_. 

.. tip:: 
    This topic was handled in issue `188 <https://github.com/Deltares/ra2ce/issues/188>`_

Replacing NetworkX with igraph
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Currently NetworkX is used as graph framework `NetworkX <https://networkx.org/>`_.
A downside of this framework is the known poor performance.
For this reason igraph `igraph <https://igraph.org/python/>`_ has been assessed as possible replacement of NetworkX.

Characteristics NetworkX
""""""""""""""""""""""""

> - works well together with `geopandas`
> - has different classes for graphs, multigraphs and directed (multi-)graphs
> - offers many functions for graph manipulation and analysis

Reviewer: ```suggestion
- works well together with `geopandas`,
- has different classes for graphs, multigraphs and directed (multi-)graphs,
- offers many functions for graph manipulation and analysis.
```

Characteristics igraph
""""""""""""""""""""""

> - has a better performance than NetworkX
> - less intuitive to use with Python and geopandas (e.g. working with coordinate systems is less straightforward)
> - less suited for dynamic graphs (indexing needs to be redone on extension and reduction of the graph)

Reviewer: ```suggestion
- has a better performance than NetworkX,
- less intuitive to use with Python and geopandas (e.g. working with coordinate systems is less straightforward),
- less suited for dynamic graphs (indexing needs to be redone on extension and reduction of the graph).
```

Conclusion
""""""""""
NetworkX is used on many places in the code.

> This, together with the differences in implementations, will require a thorough refactoring of the application for the benefit of a significant performance improvement.

Reviewer: ```suggestion
This, together with the differences between implementations, will require a thorough refactoring of the application for the benefit of a significant performance improvement.
```

> For now it is chosen not to work on replacing NetworkX with igraph.

Reviewer: ```suggestion
Therefore, for now, it was chosen not to work on replacing NetworkX with igraph.
```

.. tip:: 

>     This topic was handled in issue `222 <https://github.com/Deltares/ra2ce/issues/222>`_

Reviewer: I'd rather see this right after the first paragraph (line 21 + 1 empty space).
